Dusekpe2

dusekpetr opened the pull request:
Opraveny veci ze second screen, aplikace by mela byt funkcni.

dusekpetr committed:
Second screen

dusekpetr committed:
Repaired CreditViewController, added function for login to website, localizations and constants. Imported JGHud for showing progress in loading data, SwiftSoup for HTML parsing and KeyChainWrapper for saving users username and password. Application should be working now correctly.

janmisar commented on Podfile:
```diff
@@ -8,5 +7,8 @@ target 'isic_kredit' do
   pod 'Alamofire'
+  pod 'SwiftKeychainWrapper'
+  pod 'SwiftSoup'
+  pod 'JGProgressHUD'
```
Nepoužívej pody bez určení verze. Vysvětloval jsem to na cviku, proč je to dobrý.

dusekpetr replied:
Rad bych tam ty verze doplnil ale bohuzel nemuzu dohledat

janmisar commented on isic_kredit/ColorConstants.swift:
```diff
@@ -0,0 +1,16 @@
+    static let BackgroundColor = UIColor(red:0.27, green:0.60, blue:0.72, alpha:1.0)
+    static let LabelsColor =  UIColor(red:0.70, green:0.83, blue:0.94, alpha:1.0)
+    static let TextFieldColor =  UIColor(red:0.22, green:0.51, blue:0.63, alpha:1.0)
+    static let LoginBackgroundColor = UIColor(red:0.57, green:0.86, blue:1.00, alpha:1.0)
```
Proměnný píšem s malým na začátku, neva že jsou static - ty taky.

dusekpetr replied:
Ano, nevim proc sem to psal velkyma 😄

janmisar commented on isic_kredit/Info.plist:
```diff
@@ -16,6 +18,17 @@
+			<string>Editor</string>
+			<key>CFBundleURLSchemes</key>
+			<array>
+				<string>isic_konto</string>
```
K čemu potřebujem URL schéma? 🤔

dusekpetr replied:
To jsem zkousel nejak udelat ten TodayExtension ale uz je to pryc

janmisar commented on isic_kredit/ViewControllers/CreditViewController.swift:
```diff
@@ -0,0 +1,255 @@
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        if KeychainWrapper.standard.string(forKey: "username")==nil && KeychainWrapper.standard.string(forKey: "password")==nil{
```
Práci s keychainem by bylo vhodný wrapnout do nějaký další třídy stejně jako další logiku, ne to takhle střelit rovnou do VC. Pracuješ s tím na dalších 3 místech a všude kopíruješ string `"username"`, to je náchylný na chyby.

dusekpetr replied:
Je hotova trida KeyChain ktera to resi

janmisar commented on isic_kredit/ViewControllers/CreditViewController.swift:
```diff
@@ -0,0 +1,255 @@
+        self.hud.show(in: self.view)
+        
+        //Request website
+        Alamofire.request("https://agata.suz.cvut.cz/secure/index.php").responseString { agataResponse in
```
Tenhle cell pekelnej chain patří do zvláštní třídy - nějaká api vrstva. Stejně tak by to mělo být rozdělené na nějaké jednotlivé funkce, v tomhle se nikdo nikdy nevyzná 😞

dusekpetr replied:
Trida DataManager, chain je rozdelenej do nekolika funkci

janmisar commented on isic_kredit/ViewControllers/CreditViewController.swift:
```diff
@@ -0,0 +1,255 @@
+            if let agataResponsed = agataResponse.response{
+                
+                //We are not logged in
+                if (agataResponsed.url)!.absoluteString.contains("eduid") {
```
Na dokumentační komentář k `reloadButtonTapped` ses klidně mohl vykašlat, to vím hned co dělá už z názvu metody, ale třeba tady mohl být klidně kilometrovej komentář co očekáváme že response URL bude obsahovat, proč v tom hledám zrovna nějaký `eduid` atd. Stejně tak u zbytku tohodle strašlivýho chainu.

dusekpetr replied:
Kod jsem tentokrat okomentoval vic, doufam ze to bude lepsi

janmisar commented on isic_kredit/ViewControllers/CreditViewController.swift:
```diff
@@ -0,0 +1,255 @@
+            let tbody: Element = try agataWebSite.select("tbody").first()!
+            let td: Element = try tbody.select("td").array()[4]
+            let arrayOfMoneyStrings = td.ownText().split(separator: ",")
+            self.moneyLabel.text = "\(arrayOfMoneyStrings[0]) \(NSLocalizedString("CZK", comment: ""))"
```
Tady to bude crashovat, nemáš nikde check, že v tom poli něco je. A to si piš, že se stane mnohokrát, že tam nebude nic. Parsuješ HTML bez jakýkoliv dokumentace a příslibu, že tam něco bude nebo nebude, to musíš všechno ošetřit.

janmisar commented on isic_kredit/ViewControllers/CreditViewController.swift:
```diff
@@ -0,0 +1,255 @@
+            let arrayOfMoneyStrings = td.ownText().split(separator: ",")
+            self.moneyLabel.text = "\(arrayOfMoneyStrings[0]) \(NSLocalizedString("CZK", comment: ""))"
+        } catch{
+        }
```
Hezkej catch 😕

janmisar commented on isic_kredit/ViewControllers/SettingsViewController.swift:
```diff
@@ -0,0 +1,106 @@
+        navigationController?.setNavigationBarHidden(false, animated: false)
+        navigationController?.navigationBar.barTintColor = UIColor(red:0.27, green:0.60, blue:0.72, alpha:1.0)
+        navigationController?.navigationBar.tintColor = .white
+        navigationController?.navigationBar.titleTextAttributes = [NSAttributedString.Key.foregroundColor: UIColor.white]
```
Stylování navbaru nepatří do viewWillAppear jednoho z controllerů, který jsou na stacku, pokud k tomu není nějakej zvláštní důvod, což tady není. Stylovat třeba v appdelegate přes appearance proxy (ukazovali jsme) nebo při initu toho navigation controlleru.

dusekpetr replied:
Opraveno

janmisar commented on isic_kredit/cs.lproj/Localizable.strings:
```diff
@@ -0,0 +1,18 @@
+"Password" = "Heslo";
+"Login" = "Přihlásit se";
+"User" = "Uživatel";
+"Account balance" = "Na účtu máte";
```
Takovýhle klíče jsou zlo, ale to neřeš, tomu jsme se na cvikách moc nevěnovali. Vysvětlím klidně při odevzdání, kdyžtak se připomeň.

janmisar commented:
Komentáře k PR se neřeší tak, že PR zavřeš, v rámci jednoho commitu to opravíš, dopíšeš zbytek celý appky a pošleš novej 😐 Nehledě na to, že v zadání jsem psal tučně "posílej průběžně pull requesty", což celá appka ve dvou commitech úplně není teda.

dusekpetr committed:
Repaired code

Make class KeyChain for wraping KeyChains, error classes with messages for failing. Requests are now in DataManager in more functions.

janmisar commented on isic_kredit/ViewControllers/CreditViewController.swift:
```diff
@@ -104,133 +102,29 @@ class CreditViewController: UIViewController {
-        }
+        dataManager.finalData(success: { response in
+                                self.setMoneyValue(value: response)
+                                 self.hudSuccess(text: NSLocalizedString("Loaded", comment: ""))
```
Nechceme žádnej hud success, user chce vidět číslo a nezajímá ho, že se něco načetlo. Zmizelo loading kolečko, tak jasně že se to načetlo.

janmisar commented on isic_kredit/ViewControllers/CreditViewController.swift:
```diff
@@ -36,7 +34,7 @@ class CreditViewController: UIViewController {
         super.viewWillAppear(animated)
         
-        if KeychainWrapper.standard.string(forKey: "username")==nil && KeychainWrapper.standard.string(forKey: "password")==nil{
+        if KeyChain().getUsername() == "" && KeyChain().getPassword() == ""{
```
Proč děláš pokaždý novou instanci KeyChain? Co třeba singleton pattern? 😏

janmisar reviewed (commented):
Stále chybí readme file, odpověď na verze v podfilu mě moc neuspokojila. Co třeba `cocoapods.org`?

dusekpetr committed:
Create README.md

dusekpetr committed:
Update README.md

dusekpetr committed:
Update README.md

dusekpetr committed:
Requested changes

KeyChain is now singleton, removed success hud, it just dismissing after successful load. Podfile has versions

dusekpetr committed:
Merge branch 'dusekpe2' of https://github.com/dusekpe2/ISICkonto into dusekpe2

dusekpetr committed:
Update README.md
